File: frontend/src/utils/api/authService.ts
import { VkResponse } from "../VkService";
import { ENDPOINTS, STORAGE_KEYS } from "./config";
import { httpClient } from "./httpClient";
import { TokenResponse } from "./types";

/**
 * Authentication service for handling user authentication
 */
export class AuthService {
  /**
   * Exchange VK OAuth token for application token
   * @param authData VK authentication response
   * @returns Application token
   */
  static async exchangeToken(authData: VkResponse): Promise<string> {
    const response = await httpClient.post<TokenResponse>(
      ENDPOINTS.AUTH,
      authData,
      { requiresAuth: false },
    );

    if (response.error) {
      console.error("Authentication error:", response.error);
      throw new Error(response.error.message);
    }

    if (!response.data?.access_token) {
      throw new Error("No access token received");
    }

    // Store token in localStorage
    localStorage.setItem(STORAGE_KEYS.ACCESS_TOKEN, response.data.access_token);

    return response.data.access_token;
  }

  /**
   * Get the current authentication token
   * @returns The current token or null if not authenticated
   */
  static getToken(): string | null {
    return localStorage.getItem(STORAGE_KEYS.ACCESS_TOKEN);
  }

  /**
   * Check if the user is authenticated
   * @returns True if authenticated, false otherwise
   */
  static isAuthenticated(): boolean {
    return !!this.getToken();
  }

  /**
   * Log out the current user
   */
  static logout(): void {
    localStorage.removeItem(STORAGE_KEYS.ACCESS_TOKEN);
  }
}

File: backend/api-gateway/app/services/auth_service.py

from requests import Response
import requests as rq


class AuthService: 
    def __init__(self, url: str, headers):
        self._url: str = url + "/auth"
        self._headers = headers

    def register(self, data) -> Response:
        return rq.post(self._url + "/register", data=data, headers=self._headers)

    def login(self, data: dict[str, object]) -> Response:
        return rq.post(self._url + "/login", data=data, headers=self._headers)

    def refresh(self, token: str) -> Response:
        return rq.post(self._url + "/token/refresh", json={"RefreshToken": token}, headers=self._headers)

    def check(self, token: str) -> Response:
        return rq.get(self._url + "/check", params={"accessToken": token}, headers=self._headers)

    def service_token(self, group_id, data) -> Response:
        return rq.post(self._url + f"/groups/{group_id}/service-token/generate", data=data, headers=self._headers)

    def logout(self) -> Response:
        return rq.delete(self._url + f"/logout", headers=self._headers)
    
    